### ISSUE TITLE
ErrorException

### ISSUE THREAD

**Issue by DVendy** [akeneo-labs/spreadsheet-parser#12]
I got this when trying to import excel

ErrorException in Archive.php line 117: 
unlink(C:\Windows\Temp\xlsFE42.tmp\xl\sharedStrings.xml): Permission denied

**Comment by DVendy** [comment]
There is a problem with Archive.

**Comment by antoineguigan** [comment]
Hi,

If the file really exists, this must be a problem with your Windows configuration... I personally don't have access to a Windows system, but would gladly accept any patch, if this is fixable without removing the ability to delete temp files.

**Comment by DVendy** [comment]
Yeah I think so., I'm using it inside Laravel.
I edited the temp path to storage folder inside laravel. Then, I deleted it manually after operation inside controller. And it's successfully deleted.

**Comment by TarikAmine** [comment]
Hi, i have the same problem cited above and with on the same file, Windows can't unlink files who haven't been closed properly, i'm at loss in which part of the library the file is opened so if anyone can give any pointers on how to fix it?

**Comment by haakym** [comment]
@DVendy 

Could you explain what you did when you said this:

> Yeah I think so., I'm using it inside Laravel.
> I edited the temp path to storage folder inside laravel. Then, I deleted it manually after operation inside controller. And it's successfully deleted.

I'm also using Laravel and would love to get this working!

Thanks.

**Comment by DVendy** [comment]
@haakym , Basically, windows wont let you delete folder inside Windows/temp.,
So what I did is:
-Create a temp folder inside laravel's storage folder with permission (0777)
-Change where to put the excel's temp file
-After any function done, delete this folder

Here's the modified Archive.php
https://github.com/DVendy/GJM/blob/master/vendor_edit/Archive.php#L55

And @antoineguigan , would you fix this? because this library is way faster than any other, & I'm using it in 4 of my project., and I'm happy using it., :)

**Comment by haakym** [comment]
@DVendy Thanks for your response and explaining the issue.

I tried modified the constructor in Archive.php, but ran into the same issue:

```
unlink(C:\wamp\www\my-project\storage\temp\xlsBFBA.tmp\xl\sharedStrings.xml): Permission denied
```

Strangely when I run the same code in the windows command line it works just fine, for example:

```
php -r "unlink('C:\wamp\www\my-project\storage\temp\xlsBFBA.tmp\xl\sharedStrings.xml');"
```

Also, the command line terminal doesn't appear to be running as administrator.

Is it failing because the file is still open when attempting to `unlink()`, as @TarikAmine mentioned?

Any ideas on what I can do to fix this? Thank you.

**Comment by DVendy** [comment]
@haakym , u noticed at line 107 I immediately return the function?
Because it seems that during execution, the file is in use.,

Now look at line 774 here
https://github.com/DVendy/GJM/blob/master/app/Http/Controllers/BackController.php#L774
After operation, I manually delete the temp folder.,

**Comment by haakym** [comment]
Ah okay, I'll give this approach a try. Thanks for your help so far.

**Comment by TarikAmine** [comment]
Thanks for fix, i as well opt for this library since its the fastest solution, to work around the issue i ended up using Homestead for dev environment.  ill try the fix on Monday.

**Comment by AndrewPenry** [comment]
@antoineguigan From the manual: "The destructor method will be called as soon as there are no other references to a particular object, or in any order during the shutdown sequence." The issue is that the Archive destructor is being called before the destructor for \XMLReader, so the xml files are still open when the unlink happens. Windows won't unlink an open file. To fix this, you would need to make sure that the \XMLReader::close is explicitly called in several places. It may even take making sure that everything is destroyed by explicitly unsetting them in the Archive destructor.

I'm having the same issue on my local Windows machine, but since my production is on linux, it's not high priority for me.

**Comment by stof** [comment]
Fixed by #38

**Comment by patrikx3** [comment]
Hello!

I could only make it work in Windows using the delete via a register_shutdown_function.
Thanks.

![image](https://cloud.githubusercontent.com/assets/4770180/18304601/86e0d7bc-74e3-11e6-9cc2-5731eb1b7ff5.png)
